extensions/resource-deployment/src/services/tools/kubeCtlTool.ts

```diff
@@ -50,8 +49,11 @@ export class KubeCtlTool extends ToolBase {
 		let version: SemVer | undefined = undefined;
 		if (output) {
 			const versionJson: KubeCtlVersion = JSON.parse(output);
-			// kubectl version output might contain '+' character in the minor version, e.g. 16+, we have to remove it to make it a valid semantic version string.
-			version = new SemVer(`${versionJson.clientVersion.major}.${versionJson.clientVersion.minor.replace(/\+/g, '')}.0`);
+			if (versionJson && versionJson.clientVersion && versionJson.clientVersion.gitVersion) {
+				version = new SemVer(versionJson.clientVersion.gitVersion);
```

Comment:
what happens when versionJson.clientVersion.gitVersion is not a valid string that SemVer can pass? Does it throw? If it does throw then we are good. But if it returns some sentinel value then we may have to handle that as well.

Reply:
![image](https://user-images.githubusercontent.com/13777222/88115573-4652eb80-cb6b-11ea-97c0-8b892d296698.png)

Reply:
semver will throw error.